### PR TITLE
AM2R: Logic updates for A0-A3

### PR DESCRIPTION
**Pull request by JeffGainsNGames**
#### Main Caves

	- Added: Octroll Lava Pool - Crossing with only power grip

#### Golden Temple

	- Added: 1-Orb Hallway - now has wave beam method to shoot Orb
	- Added: Golden Temple Exterior - Speedbooster + HJ to exterior alpha nest
	- Changed: Golden Temple Exterior - corrected climb from morph glide to mid-air morph
	- Added: Guardian Storage - Spring Ball method

#### Hydro Station

	- Changed: Hydro Station Exterior - Updated morph glides to right missile tank with more appropriate requirements

#### Industrial Complex

	- Changed: Exterior Maze - now includes supers, added `not entrance rando` to movement as this causes issues coming from the bottom
	- Changed: Spazer Beam Access - removed knowledge and lowered shinespark level to be in line with others
	- Added: Upper Factory Intersection - Shinespark method for climbing to save from spazer access

### CODE REVIEW

File: randovania/games/am2r/json_data/Golden Temple.txt [hunk 1]
```diff
@@ -704,6 +706,9 @@ Extra - minimap_data: [{'x': 62, 'y': 12}, {'x': 63, 'y': 12}, {'x': 64, 'y': 12
           Mid-Air Morph (Beginner) or Tunnel Climb
   > Pickup (Energy Tank)
       After Area 1 - Golden Temple 1 Orb Puzzle
+  > Event - 1Orb
```
Comment (Miepee) [inline randovania/games/am2r/json_data/Golden Temple.txt:709]: assuming you hit the event with wave, how do you "get out"? from what i can see, itd still logically expect boombs for you i the current state

File: randovania/games/am2r/json_data/Hydro Station.txt [hunk 2]
```diff
@@ -103,7 +103,16 @@ Extra - minimap_data: [{'x': 29, 'y': 18}, {'x': 29, 'y': 19}, {'x': 29, 'y': 20
...
+          All of the following:
+              # Morph Glide
+              Morph Ball
+              Any of the following:
+                  All of the following:
+                      # Without HJ
+                      Morph Glide (Expert)
+                      Power Grip or Walljump (Advanced)
+                  # With HJ
+                  Hi-Jump Boots and Morph Glide (Advanced)
```
Comment (Miepee) [inline randovania/games/am2r/json_data/Hydro Station.txt:115]: The glide requirements dont seem right here at all. 
Without PG or WJ, its an expert
with PG or WJ its intermediate
with HJ its beginner

File: randovania/games/am2r/json_data/Industrial Complex.txt [hunk 3]
```diff
@@ -495,6 +495,8 @@ Extra - minimap_data: [{'x': 64, 'y': 22}, {'x': 64, 'y': 23}, {'x': 64, 'y': 24
 > Horizontal Dock to Spazer Beam Chamber Access; Heals? False
   * Layers: default
   * 4-Tiles High Dock to Spazer Beam Chamber Access/Horizontal Dock to Upper Factory Intersection
+  > Dock to Upper Factory Save Station
+      Speed Booster and Shinesparking Tricks (Beginner) and Walljump (Beginner) and Disabled Entrance Rando
```
Comment (Miepee) [inline randovania/games/am2r/json_data/Industrial Complex.txt:499]: isnt this also possible with PG / HJ / Space rathern than requiring WJ?

File: randovania/games/am2r/json_data/Golden Temple.txt (hunk 1 above)
Comment (JeffGainsNGames) [reply randovania/games/am2r/json_data/Golden Temple.txt:709]: It seems the I missed the return path when working on this, intended to have the same requirements as activating with wave (`wave & knowledge`) to get out. Will correct.

File: randovania/games/am2r/json_data/Hydro Station.txt (hunk 2 above)
Comment (JeffGainsNGames) [reply randovania/games/am2r/json_data/Hydro Station.txt:115]: I have to say the `No HJ + PG or WJ` feels greater than intermediate due to the distance. See the below image for an intermediate Morph Glide we have already.
image


Agree with the rest.

File: randovania/games/am2r/json_data/Industrial Complex.txt (hunk 3 above)
Comment (JeffGainsNGames) [reply randovania/games/am2r/json_data/Industrial Complex.txt:499]: Could be HJ, space is already covered in the other transitions between room bottom->mid->upper. PG could also help reach the platform to spark straight up to the save.

File: randovania/games/am2r/json_data/Hydro Station.txt (hunk 2 above)
Comment (Miepee) [reply randovania/games/am2r/json_data/Hydro Station.txt:115]: The one from a3 is actually a bit tigher if i remember correctly
The one in a2 is *very* easy, as you can pretty much just spam morph

**Review by Miepee — APPROVED** [review]

**Pull request merged**